source/common/grpc/grpc_web_filter.cc

```diff
@@ -0,0 +1,130 @@
+#include "common/grpc/grpc_web_filter.h"
+
+#include <arpa/inet.h>
+
+#include "common/common/base64.h"
+#include "common/http/headers.h"
+
+namespace Envoy {
+namespace Grpc {
+
+// Bit mask denotes a trailers frame of gRPC-Web.
```

Comment:
I'm going to add another bit mask 0b00000001 for compression. So it's not necessary to mention the compression here once it gets in.